pkg/nodeconfig/nodeconfig.go

```diff
@@ -180,12 +191,13 @@ func (nc *nodeConfig) Configure() error {
 		if err := nc.setNode(false); err != nil {
 			return errors.Wrap(err, "error getting node object")
 		}
+
+		// Version annotation is the indicator that the node was fully configured by this version of WMCO, so it should
+		// be added at the end of the process.
 		nc.addVersionAnnotation()
-		nc.addPubKeyHashAnnotation()
-		nc.addAdditionalAnnotations()
-		node, err := nc.k8sclientset.CoreV1().Nodes().Update(context.TODO(), nc.node, meta.UpdateOptions{})
+		node, err = nc.k8sclientset.CoreV1().Nodes().Update(context.TODO(), nc.node, meta.UpdateOptions{})
 		if err != nil {
-			return errors.Wrap(err, "error updating node labels and annotations")
+			return errors.Wrapf(err, "error updating annotations on node %s", nc.node.GetName())
```

Comment:
nit: `error updating version annotation on node...`
Differentiating the error message from earlier annotation additions would give us a better idea where in the configuration process something failed